Sources/MapboxMaps/Location/LocationProducer.swift

```diff
@@ -84,15 +74,20 @@ internal final class LocationProducer: LocationProducerProtocol {
                 }
                 locationProvider.startUpdatingLocation()
                 locationProvider.startUpdatingHeading()
+                updateHeadingOrientationIfNeeded()
+                startUpdatingInterfaceOrientation()
             } else {
                 locationProvider.stopUpdatingLocation()
                 locationProvider.stopUpdatingHeading()
+                stopUpdatingInterfaceOrientation()
             }
         }
     }
 
+    private var _ignoreLocationProviderUpdate = false
```

Comment:
This is needed because `LocationProvider` is of `Any` type so it behaves as a struct - setting `locationProvider.headingOrientation = headingOrientation` causes property observes of `locationProvider` property gets invoked.